src/main/java/org/codehaus/plexus/util/SelectorUtils.java

```diff
@@ -251,23 +251,38 @@ public static boolean matchPath( String pattern, String str, boolean isCaseSensi
 
     public static boolean matchPath( String pattern, String str, String separator, boolean isCaseSensitive )
     {
-        if ( isRegexPrefixedPattern( pattern ) )
+        /* 2022-03-30: Fixed: Caller parameter in a public method should be handled immutable. */
+        String localPattern = pattern;
+        if ( isRegexPrefixedPattern( localPattern ) )
         {
-            pattern =
-                pattern.substring( REGEX_HANDLER_PREFIX.length(), pattern.length() - PATTERN_HANDLER_SUFFIX.length() );
+            localPattern =
+                localPattern.substring( REGEX_HANDLER_PREFIX.length(), localPattern.length() - PATTERN_HANDLER_SUFFIX.length() );
 
-            return str.matches( pattern );
+            return str.matches( localPattern );
         }
         else
         {
-            if ( isAntPrefixedPattern( pattern ) )
+            if ( isAntPrefixedPattern( localPattern ) )
             {
-                pattern = pattern.substring( ANT_HANDLER_PREFIX.length(),
-                                             pattern.length() - PATTERN_HANDLER_SUFFIX.length() );
+                localPattern = localPattern.substring( ANT_HANDLER_PREFIX.length(),
+                                             localPattern.length() - PATTERN_HANDLER_SUFFIX.length() );
             }
```

Comment:
Replace with
```
String localPattern = isAntPrefixedPattern( pattern ) ? pattern.substring( ... ) : pattern;
```

Reply:
@gnodet: Thx for reviewing. I modified the code as mentioned.